language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/MakeConstructPublicCodeAction.java

```diff
@@ -0,0 +1,123 @@
+        if (!DIAGNOSTIC_CODE.contains(diagnostic.diagnosticInfo().code()) || context.currentSyntaxTree().isEmpty()
+                || context.currentSemanticModel().isEmpty()) {
+            return Collections.emptyList();
+        }
+        Range diagnosticRange = new Range(CommonUtil.toPosition(diagnostic.location().lineRange().startLine()),
+                CommonUtil.toPosition(diagnostic.location().lineRange().endLine()));
+        NonTerminalNode nonTerminalNode = CommonUtil.findNode(diagnosticRange, context.currentSyntaxTree().get());
+        Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(nonTerminalNode);
+        if (symbol.isEmpty() || symbol.get().getModule().isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Optional<Project> project = context.workspace().project(context.filePath());
+        if (project.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Optional<Path> filePath = CommonUtil.getFilePathForSymbol(symbol.get(), project.get(), context);
+        if (filePath.isEmpty() || context.workspace().syntaxTree(filePath.get()).isEmpty()) {
+            return Collections.emptyList();
+        }
+        URI uri = filePath.get().toUri();
+
+        Optional<NonTerminalNode> node = CommonUtil.findNode(symbol.get(),
+                context.workspace().syntaxTree(filePath.get()).get());
+        if (node.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Position position = getStartPosition(node.get());
+        Range range = new Range(position, position);
+        String editText = SyntaxKind.PUBLIC_KEYWORD.stringValue() + " ";
+        TextEdit textEdit = new TextEdit(range, editText);
+        List<TextEdit> editList = List.of(textEdit);
+        String commandTitle = String.format(CommandConstants.MAKE_CONSTRUCT_PUBLIC, symbol.get().getName().orElse(""));
+        return List.of(createCodeAction(commandTitle, editList, uri.toString(), CodeActionKind.QuickFix));
+    }
+
+    private static Position getStartPosition(Node node) {
+        Position startPosition = new Position();
+        SyntaxKind typeKind = node.kind();
+
+        if (typeKind.equals(SyntaxKind.TYPE_DEFINITION)) {
+            TypeDefinitionNode typeDefinitionNode = (TypeDefinitionNode) node;
+            startPosition = CommonUtil.toPosition(typeDefinitionNode.typeKeyword().lineRange().startLine());
+        }
+        if (typeKind.equals(SyntaxKind.CLASS_DEFINITION)) {
+            ClassDefinitionNode classDefinitionNode = (ClassDefinitionNode) node;
+            if (!classDefinitionNode.classTypeQualifiers().isEmpty()) {
+                startPosition = CommonUtil.toPosition(classDefinitionNode.classTypeQualifiers().get(0)
+                        .lineRange().startLine());
+            } else {
+                startPosition = CommonUtil.toPosition(classDefinitionNode.classKeyword().lineRange().startLine());
+            }
+        }
```

Comment:
If we only support classes and types at the moment, instead of returning the original position if none of the if clauses are matched, shall we return `Optional`? This will ensure do not allow any mishaps in the future as well. WDYT?
Also, try to return from the if clauses.